syft/execution/plan.py

```diff
@@ -775,6 +800,13 @@ def unbufferize(worker: AbstractWorker, protobuf_plan: PlanPB) -> "Plan":
         if protobuf_plan.description:
             plan.description = protobuf_plan.description
 
+        # Replace non-instanciated placeholders from plan.placeholders by instanciated placeholders
+        # from state.state_placeholders
+        # TODO this definitely isn't the right strategy. Maybe state shouldn't contain
+        # instanciated placeholders but values directly?
```

Comment:
I think the approach is okay, but wonder if this makes sense to extract as a method since it happens in multiple places.